### PR TITLE
QAE-260 - E2E test case for Transparent header enable on complete website

### PR DESCRIPTION
**Pull request by SimranChougule1**
### Description

Added test case for Transparent header enable on complete website

### Screenshots

https://share.getcloudapp.com/NQu7249D

### How has this been tested?

npm run test:e2e:interactive tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js

### Checklist:
- [ ] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import { setCustomize } from '../../../../utils/customize';
+describe( 'transparent header in the customizer', () => {
+	it( 'disable on archive should apply corectly', async () => {
+		const Disableonarchive = {
+			'transparent-header-enable': 1,
+			'transparent-header-disable-archive': {
+				position: 'absolute',
+				left: 0,
+				right: 0,
+
```
Comment (premanshup) [inline tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js:11]: @SimranChougule1 This is not an array, this should be either 1 or 0, can you please confirm the same?

**Review by premanshup — CHANGES_REQUESTED** [review]

File: tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		} ).cssValueToBe(
+			`${ Disableonarchive[ 'transparent-header-disable-archive' ].position }` );
+	} );
+	it( 'disable on latest posts page should apply corectly', async () => {
+		const Disableonlatestpostpage = {
+			'transparent-header-enable': 1,
+			'transparent-header-disable-latest-posts-index': {
+				position: 'absolute',
+				left: 0,
+				right: 0,
```
Comment (premanshup) [inline tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js:32]: @SimranChougule1 This as well.

File: tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+			`${ Disableonlatestpostpage[ 'transparent-header-disable-latest-posts-index' ].position }` );
+	} );
+	it( 'disable on pages should apply corectly', async () => {
+		const Disableonpages = {
+			'transparent-header-enable': 1,
+			'transparent-header-disable-page': {
+				position: 'absolute',
+				left: 0,
+				right: 0,
+
```
Comment (premanshup) [inline tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js:59]:
```suggestion
				position: 'absolute',
				left: 0,
				right: 0,

```
This too

File: tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+	} );
+	it( 'disable on posts should apply corectly', async () => {
+		const Disableonposts = {
+			'transparent-header-enable': 1,
+			'transparent-header-disable-posts': {
+				position: 'absolute',
+				left: 0,
+				right: 0,
+
+			},
```
Comment (premanshup) [inline tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js:84]:
```suggestion
			'transparent-header-disable-posts': {
				position: 'absolute',
				left: 0,
				right: 0,

			},
```

File: tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js (hunk 1 above)
Comment (SimranChougule1) [reply tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js:11]: Okay I'll check

Comment (SimranChougule1) [reply tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js:11]: @imnavanath suggested me to add this array

File: tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
+import { createURL,	createNewPost, publishPost } from '@wordpress/e2e-test-utils';
```
Comment (bsfsnehal) [inline tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js:1]: required change in "publish-post" util path

File: tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+import { setCustomize } from '../../../../utils/customize';
+describe( 'transparent header in the customizer', () => {
```
Comment (bsfsnehal) [inline tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js:3]: 1st letter should be capital

File: tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+	it( 'enable on archive, 404 and search page should apply corectly', async () => {
+		const Disableonarchive = {
```
Comment (bsfsnehal) [inline tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js:5]: constant name should be in camelCase

File: tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js (hunk 5 above)
Comment (SimranChougule1) [reply tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js]: Done

File: tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js (hunk 6 above)
Comment (SimranChougule1) [reply tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js]: Done

File: tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js (hunk 7 above)
Comment (SimranChougule1) [reply tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js]: Done

File: tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js (hunk 1 above)
Comment (SimranChougule1) [reply tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js]: Done

**Review by bsfsnehal — APPROVED** [review]

**Review by GauravKhupse — APPROVED** [review]

File: tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+		} );
+		await page.waitForSelector( '.ast-theme-transparent-header' );
+		const disableArchive = await page.$eval( '.ast-theme-transparent-header', ( element ) => element.getAttribute( '.ast-theme-transparent-header #masthead' ) );
+		await expect( disableArchive ).toBeNull( );
+
+		await setCustomize( disableOnArchive );
+		await page.goto( createURL( '/12' ), {
+			waitUntil: 'networkidle0',
+		} );
+		await page.waitForSelector( '.ast-theme-transparent-header' );
```
Comment (GauravKhupse) [inline tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js:22]: 404 page is not showing in the preview.

Comment (bsfsnehal) [reply tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js:22]: @SimranChougule1 can you please verify this?

Comment (SimranChougule1) [reply tests/e2e/specs/customizer/header-builder/transparent-header/enable-on-complete-website.test.js:27]: Resolved

**Review by premanshup — APPROVED** [review]

**Pull request merged**